Add diffusion theory and associated kernels to Theory page on Moltres website (#270)
arfc/moltres: doc/content/getting_started/theory.md

## Multigroup Neutron Diffusion

The neutron diffusion equation is an approximation to the Boltzmann transport equation, and is derived by taking the zeroth and first moment with respect to $\hat\Omega$, the direction of neutron travel. 

> A full derivation of the diffusion equation will not be presented here for conciceness. Importantly, when taking the first moment of the transport equation the angular flux is assumed to be linearly anisotropic. 

Reviewer: ```suggestion
A full derivation of the diffusion equation will not be presented here for conciseness. Importantly, when taking the first moment of the transport equation the angular flux is assumed to be linearly anisotropic. 
```

The approximations reduce the phase space through the elimination of angular dependence, but the resulting equation also has reduced fidelity when compared to the transport equation, particularly in regions where the neutron flux has strong angular dependence. 
A non-exhaustive list of regions in which the neutron flux strongly depends on angle is near material interfaces between materials with highly dissimilar neutronic properties, within strong absorbers, and within near-void regions.

> As a dertiministic method, the neutron diffusion method also requires discretization of the continuous energy dependence into energy groups consisting of non-overlapping, finite energy ranges across the entire energy spectrum. 

Reviewer: ```suggestion
As a deterministic method, the neutron diffusion method also requires discretization of the continuous energy dependence into energy groups consisting of non-overlapping, finite energy ranges across the entire energy spectrum. 
```

This energy discretization creates a system of equations referred to as the multigroup neutron diffusion equations: 

!equation
\frac{1}{v_g} \frac{\partial\phi_g}{\partial t}-\nabla \cdot D_g \nabla \phi_g +\Sigma^R_{g} \phi_g =\sum^G_{g \neq g'} {\Sigma^s_{g'\rightarrow g} \phi_{g'}}+ \chi^p_{g} \sum^G_{g'=1} {\left(1- \beta \right) \nu_{g'} \Sigma^f_{g'}\phi_{g'} }+\chi^d_{g} \sum^I_i {\lambda_i C_i}

> where the precursor distributions are governed by:

Reviewer: ```suggestion
The delayed neutron precursor distributions are governed by:
```

!equation
\frac{\partial C_i}{\partial t}=\sum^G_{g'=1}{\beta_i \nu \Sigma^f_{g'}\phi_{g'}}-\lambda_i C_i-\vec{u} \cdot \nabla C_i

> 

Reviewer: ```suggestion
```
Remove extra empty line here

## Multigroup Neutron Diffusion

The neutron diffusion equation is an approximation to the Boltzmann transport equation, and is derived by taking the zeroth and first moment with respect to $\hat\Omega$, the solid angle. 
A full derivation of the diffusion equation will not be presented here for the sake of conciceness. Importantly, when taking the first moment of the transport equation the angular flux is assumed to be linearly anisotropic. 
This assumption yields an equation with 2 fewer phase spaces, but also has reduced fidelity when compared to the transport equation, particularly in regions in which the neutron flux has strong angular dependence. 
A non-exhaustive list of regions in which the neutron flux strongly depends on angle is near material interfaces, external boundaries, strong absorbers, and weak scaterrers.

The general neutron diffusion equation has exceedingly few analytical solutions and thus, for real-world problems, is solved with numerical methods. 
A common method utilized is [Finite Element Method](https://mooseframework.inl.gov/help/faq/what_is_fem.html), which is the methodology employed in Moltres. 

> To solve the diffusion equation over complex geometries, Moltres discretizes over space, time, and energy. 
> Discretization through space is done with the use of volume meshes. 
> Discretization with respect to time is accomplished with discrete time-steps. 

Reviewer: ```suggestion
In FEM, the spatial domain is discretized into finite mesh elements while the time dependence is modeled through discrete time steps and time integration methods.
```

Finally, the energy spectrum of the scalar neutron flux is discretized into user defined bins, called energy groups. 
The energy discretization of the neutron diffusion equation is called the multi-group neutron diffusion equation and is presented below. 
Notably, there are two production terms of neutrons, the prompt fission source and delayed neutron precursor decay source. 

> The first term describes the neutrons immediately born from fission, and the second term describes the neutrons born from the radioactive decay of neutron-emitting radionuclides, commonly called delayed neutron precursors. 

Reviewer: Move these to the top of the second paragraph

|Term in Multi-Group Diffusion Equation | Associated Kernel |Definition of Term|
| - | - | - |
| [!eq](\frac{1}{v_g} \frac{\partial\phi_g}{\partial t}) | [NtTimeDerivative](https://arfc.github.io/moltres/source/kernels/NtTimeDerivative.html) | Time rate of change of energy group g |
| [!eq](- \nabla \cdot D_g \nabla \phi_g) | [GroupDiffusion](https://arfc.github.io/moltres/source/kernels/GroupDiffusion.html) | Streaming term of energy group g |
| [!eq](\Sigma^R_{g} \phi_g) | [SigmaR](https://arfc.github.io/moltres/source/kernels/SigmaR.html) | Removal from energy group g |
| [!eq](\sum^G_{g \neq g'} {\Sigma^s_{g'\rightarrow g} \phi_{g'}}) | [InScatter](https://arfc.github.io/moltres/source/kernels/InScatter.html) | In-scattering into energy group g | 
| [!eq](\chi^p_{g} \sum^G_{g'=1} {\left[1- \beta \right] \nu_{g'} \Sigma^f_{g'}\phi_{g'} } ) | [CoupledFissionKernel](https://arfc.github.io/moltres/source/kernels/CoupledFissionKernel.html) | Prompt fission neutron source |
| [!eq](\chi^d_{g} \sum^I_i {\lambda_i C_i}) | [DelayedNeutronSource](https://arfc.github.io/moltres/source/kernels/DelayedNeutronSource.html) | Delayed fission neutron source |

 
\\  

> The governing equation for the delayed neutron precursor population:

Reviewer: ```suggestion
The governing equation for the delayed neutron precursor distribution is:
```
